src/lib.rs

```diff
@@ -7,6 +7,10 @@ pub trait Merge {
     fn merge(self, other: Self) -> Self;
 }
 
+pub trait MergeMut {
```

Comment:
Dodges a necessary clone in Edge where we have a mutable reference but need an owned type. This lets us mutate in place rather